sw/host/opentitanlib/src/io/uart.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use std::io::{Read, Write};
+use std::time::Duration;
+
+/// A trait which represents a UART.
+pub trait Uart: Read + Write {
+    /// Returns the UART baudrate.  May return zero for virtual UARTs.
+    fn get_baudrate(&self) -> u32;
```

Comment:
If this can return zero for virtual UARTs, does `Option<u32>` or `Option<std::num::NonZeroU32>` make more sense?

Reply:
I guess I don't have too strong of an opinion on this other than I'd like `foo.set_baudrate(foo.get_baudrate())` to do the intuitively obvious thing.  My preference is to just return zero for fake UARTs.

Reply:
Okay, SGTM